12-multiplatform/3_tinycity/3_2_crossing/exercice/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"time"
+
+	"machine"
+)
+
+func main() {
+	car := newCarLight(machine.D2, machine.D3, machine.D4)
+	walk := newWalkLight(machine.D5, machine.D6)
+	c := newCrossing(car, walk)
+
+	for {
+		c.Switch()
+		time.Sleep(time.Second * 5)
+	}
+}
+
+type crossing struct {
+	cars          *carLight
+	walks         *walkLight
+	pedestriansGo bool
+}
+
+func newCrossing(cars *carLight, walks *walkLight) *crossing {
+	return &crossing{
+		cars:  cars,
+		walks: walks,
+	}
+}
+
+func (c *crossing) Switch() {
+	if c.pedestriansGo {
+		c.pedestriansGo = false
+		c.walks.Stop()
+		c.cars.Go()
+	} else {
+		c.pedestriansGo = true
```

Comment:
I would use `c.pedestriansGo = !c.pedestriansGo` as it's more obvious that it's toggling every cycle -- and there's less likely of a chance to mix up the true/false.

But, this approach is fine as well.